File: 2141-maximum-running-time-of-n-computers/2141-maximum-running-time-of-n-computers.java
class Solution {
    public long maxRunTime(int n, int[] batteries) {
        long totalEnergy = 0;
        for (int battery : batteries) {
            totalEnergy += battery;
        }

        Arrays.sort(batteries);

        for (int i = batteries.length - 1; i >= 0; i--) {
            if (batteries[i] > totalEnergy / n) {
                totalEnergy -= batteries[i];
                n--;
            } else {
                break;
            }
        }

        return totalEnergy / n;
    }
}

File: 2141-maximum-running-time-of-n-computers/2141-maximum-running-time-of-n-computers.cpp
class Solution {
public:
    long long maxRunTime(int n, vector<int>& batteries) {
        long long totalEnergy = 0;
        for (int b : batteries) totalEnergy += b;

        sort(batteries.begin(), batteries.end());

        for (int i = batteries.size() - 1; i >= 0; i--) {
            if (batteries[i] > totalEnergy / n) {
                totalEnergy -= batteries[i];
                n--;
            } else break;
        }

        return totalEnergy / n;
    }
};

